pkg/services/ngalert/notifier/autogen_alertmanager_test.go

```diff
@@ -164,6 +164,71 @@ func TestAddAutogenConfig(t *testing.T) {
 				},
 			}),
 		},
+		{
+			name:           "settings with custom options and nil groupBy, groupBy should inherit from parent",
+			existingConfig: configGen([]string{"receiver1"}, nil),
+			storeSettings: []models.NotificationSettings{
+				models.CopyNotificationSettings(models.NewDefaultNotificationSettings("receiver1"), models.NSMuts.WithGroupInterval(util.Pointer(1*time.Minute)), models.NSMuts.WithGroupBy()),
+			},
+			expRoute: withChildRoutes(rootRoute(), &definitions.Route{
+				Receiver:       "default",
+				ObjectMatchers: matcher(models.AutogeneratedRouteLabel, "true"),
+				Routes: []*definitions.Route{
+					withChildRoutes(basicContactRoute("receiver1"), &definitions.Route{
+						Receiver:       "receiver1",
+						ObjectMatchers: matcher(models.AutogeneratedRouteSettingsHashLabel, "dde34b8127e68f31"),
+						GroupByStr:     nil,
+						GroupInterval:  util.Pointer(model.Duration(1 * time.Minute)),
+					}),
+				},
+			}),
+		},
+		{
+			name:           "settings with nil groupBy should have different fingerprint than default groupBy",
+			existingConfig: configGen([]string{"receiver1"}, nil),
+			storeSettings: []models.NotificationSettings{
+				models.CopyNotificationSettings(models.NewDefaultNotificationSettings("receiver1"), models.NSMuts.WithGroupInterval(util.Pointer(1*time.Minute)), models.NSMuts.WithGroupBy()),
+				models.CopyNotificationSettings(models.NewDefaultNotificationSettings("receiver1"), models.NSMuts.WithGroupInterval(util.Pointer(1*time.Minute)), models.NSMuts.WithGroupBy(models.DefaultNotificationSettingsGroupBy...)),
+			},
+			expRoute: withChildRoutes(rootRoute(), &definitions.Route{
+				Receiver:       "default",
+				ObjectMatchers: matcher(models.AutogeneratedRouteLabel, "true"),
+				Routes: []*definitions.Route{
+					withChildRoutes(basicContactRoute("receiver1"), &definitions.Route{
+						Receiver:       "receiver1",
+						ObjectMatchers: matcher(models.AutogeneratedRouteSettingsHashLabel, "dde34b8127e68f31"),
+						GroupByStr:     nil,
+						GroupInterval:  util.Pointer(model.Duration(1 * time.Minute)),
+					}, &definitions.Route{
+						Receiver:       "receiver1",
+						ObjectMatchers: matcher(models.AutogeneratedRouteSettingsHashLabel, "e1f3a275a8918385"), // Different hash.
+						GroupByStr:     []string{models.FolderTitleLabel, model.AlertNameLabel},
+						GroupInterval:  util.Pointer(model.Duration(1 * time.Minute)),
+					}),
+				},
+			}),
+		},
```

Comment:
It's not necessary but a good idea, IMO. 

It's more inline with how we generate routes for other fields where nil means not overridden. More importantly though, if we change the logic so that nil `GroupBy` gets normalized to the `grafana_folder, alertname` then it will cause fingerprint changes for all rules that override something but don't override the `GroupBy`.